### DISCUSSION TITLE
Failed to fetch metric data

### DISCUSSION THREAD

**Discussion by AnDBAIdewe** [cybertec-postgresql/pgwatch2#673] [Q&A] [answered]
After upgrading a db from PG14 to 15, I get the following error in PgWatch:
2023/08/07 07:51:55 ERRO MetricGathererLoop: Failed to fetch metric data for [an_dev_an_dev:db_stats]: pq: function pg_backup_start_time() does not exist

DB is  PostgreSQL 15.2
pgwatch-t01 ~]# docker ps -a
IMAGE
cybertec/pgwatch2:1.10.0

According to the doc, this version should work for PG15? It looks like the PG14 metrics are still being used.

**Comment by pashagolub** [comment, marked as answer]
According to the PostgreSQL [chagelog](https://www.postgresql.org/docs/15/release-15.html):

>Remove long-deprecated [exclusive backup mode](https://www.postgresql.org/docs/15/continuous-archiving.html#BACKUP-BASE-BACKUP) (David Steele, Nathan Bossart)
If the database server stops abruptly while in this mode, the server could fail to start. The non-exclusive backup mode is considered superior for all purposes. Functions pg_start_backup()/pg_stop_backup() have been renamed to pg_backup_start()/pg_backup_stop(), and the functions pg_backup_start_time() and pg_is_in_backup() have been removed.

You need to update your metrics definition as well.

**Comment by AnDBAIdewe** [comment]
Thanks! Works now. System engineer who did the upgrade forgot to read the full doc ;-)

**Reply by pashagolub** [reply]
Haha! Classic! :-D
